### DISCUSSION TITLE
Is it possible to do a shift+click on an element with Vimium-C ?

### DISCUSSION THREAD

**Discussion by acelticsfan** [gdh1995/vimium-c#1045] [Q&A] [answered]
By default, Jira doesn't load all of the comments for an issue.  So, a workaround for that I've used is to map a key in Vimium-C to click on the load older comments bar/button automatically.  When I press the key it finds the element and then clicks on it.

By using, this workaround it was a single keystroke to click the element and list all comments.

map c LinkHints.click direct="element"  match=".show-more-comments"

In the new Jira 9.x, the element names have changed a bit and the match is now this:
map c LinkHints.click direct="element"  match=".show-more-comment-tabpanel"

This works, but it only displays 10 comments at a time.   If you do a shift-click on the element, it will display all comments.  Is it possible to do this?  

Another way to accomplish this would be by performing the action that the click on the button bar does.  The click action invokes a URL to display the additional comments.   There are two URLs:  the first only loads 10 comments, the second loads all comments.

Is it possible to click on the element and invoke its second URL at the same time?

This is HTML associated with the "load comment bar".

`<button class="collapsed-actions collapsed-comment-tabpanel show-more-comment-tabpanel show-more-tab-items aui-button" data-fetch-mode="older" data-url="/browse/ISSUE-173375?page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel&amp;fetchMode=OLDER&amp;actionOrder=asc&amp;fromDate=2023-11-06T10%3A52%3A16.000-0800" data-url-all="/browse/ISSUE-173375?page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel&amp;fetchMode=OLDER&amp;showAll=true&amp;actionOrder=asc&amp;fromDate=2023-11-06T10%3A52%3A16.000-0800" data-load-all-message="select to load all" data-load-all-enabled="true" resolved="" aria-describedby="aui-tooltip">Load 10 older comments. <span class="shift-key-icon">Shift</span> + select to load all.</button>`

**Comment by acelticsfan** [comment]
A 3rd option would be to keep doing the click on the element until it no longer exists in the page after Jira has expanded all the comments.

**Comment by gdh1995** [comment, marked as answer]
Um, `LinkHints.activate` has no parameter to "always do a click with Shift=pressed`, but there's indeed a parameter to achieve this effect "most of time" - try `newtab="window"`:
```
map c LinkHints.click direct="element" match=".show-more-comments-tabpanel" newtab="window"
```

BTW, I'm not sure why you used `mode=37`.
- `37` is to `open an image`, but in this case the target is a standard button
- `mode: number` is not suggested although it does work now. Instead, its official usage is `mode: string`, and allowed values are: 
https://github.com/gdh1995/vimium-c/blob/64ff7776d5f83c04f76037e8a1d4ef7285c7144e/background/key_mappings.ts#L828-L847

**Comment by acelticsfan** [comment]
mode=37 was a mistake.  Thanks for getting back and the explanation.  I'll give it a try!!

Appreciate the help;  Vimium-C is great!

**Comment by acelticsfan** [comment]
This mapping works for me.  
c -> show all comments by running twice.  For first time, Jira doesn't give option of doing shift+click.  So have to run this twice.

map C LinkHints.click direct="element" match=".show-more-comment-tabpanel" newtab="window"
map c runKey keys="C+800wait+C"

However, if it try to simplify runKey to just run it doesn't work.  

run c C,800wait,C

But, using runkey is good enough for me.  Thanks, again.

**Reply by gdh1995** [reply]
If you want to use `run xxx xxxx`, then you still need the `+` character to join two sequential keys, so you may try:
```
run c C+800wait+C
```
